scanpipe/models.py

```diff
@@ -1943,6 +1943,13 @@ def location_path(self):
         path = Path(str(self.path).strip("/"))
         return self.project.codebase_path / path
 
+    @property
+    def name_without_extension(self):
+        """Return the name of the resource without it's extension."""
+        if self.extension:
+            return self.name.replace(self.extension, "")
```

Comment:
Using replace in that context is not ideal.

Reply:
Using `.rstrip()` instead of `replace` now, that should be more precise?
Thanks!

Reply:
`.rstrip` is dangerous:
```
>>> "file.x.ext".rstrip(".ext")
'fil'
```

Reply:
Yes, using `.rpartition()` instead now. This should be safe.